src/App.tsx

```diff
@@ -0,0 +1,122 @@
+import { Button } from '@/components/ui/button';
+import { Label } from '@/components/ui/label';
+import { useQuery } from '@tanstack/react-query';
+import { useEffect, useState } from 'react';
+import { nodeUrl } from '@/utils/const';
+import { Balance } from '@/types/balances-return';
+import { Skeleton } from '@/components/ui/skeleton';
+import { Textarea } from '@/components/ui/textarea';
+import { toast } from 'sonner';
+import dictionary from '@/dictionary/en.json';
+import { HoldersForm } from '@/components/HoldersForm';
+
+const App = () => {
+  const [tokenId, setTokenId] = useState<string>('');
+  const [minAmount, setMinAmount] = useState<number | null>(null);
+  const [data, setData] = useState<Balance[]>([]);
+  const [shouldFetch, setShouldFetch] = useState(false);
+
+  const copyToClipboard = async (textToCopy: string) => {
+    if (navigator.clipboard && window.isSecureContext) {
+      await navigator.clipboard.writeText(textToCopy);
+      toast.success(dictionary.copiedToClipboard);
+      return;
+    } else {
+      const textArea = document.createElement('textarea');
+      textArea.value = textToCopy;
+
+      textArea.style.position = 'absolute';
+      textArea.style.left = '-999999px';
+
+      document.body.prepend(textArea);
+      textArea.select();
+
+      try {
+        document.execCommand('copy');
+      } catch (error) {
+        console.error(error);
+      } finally {
+        textArea.remove();
+      }
+    }
+  };
+
+  const fetchData = async (url: string) => {
+    const response = await fetch(url);
+
+    if (!response.ok) {
+      throw new Error(`${dictionary.httpError} ${response.status}`);
+    }
+
+    const data = await response.json();
+
+    setData((prevData: Balance[]) => [...prevData, ...data?.balances]);
+
+    if (data.links.next) {
+      await fetchData(`${nodeUrl}${data.links.next}`);
+    }
+
+    return data;
+  };
+
+  const { error, isFetching, isFetched, isSuccess } = useQuery({
+    enabled: shouldFetch,
+    retry: 0,
+    throwOnError: false,
+    queryKey: ['queryList'],
+    queryFn: () => fetchData(`${nodeUrl}/api/v1/tokens/${tokenId}/balances?account.balance=gte:${minAmount}&limit=100`),
+  });
+
+  useEffect(() => {
+    if (isSuccess) toast.success(dictionary.successfullyFetchedData);
+  }, [isSuccess]);
+
+  useEffect(() => {
+    if (error) {
+      toast.error(error.toString());
+    }
+  }, [error]);
```

Comment:
You can probably move toast logic into try / catch block inside `fetchData` func

Reply:
I preffer to throw all errors in fetchData and use error from useQuery to display it. I think it's more efficient way